tests/integration/.golangci.yaml

```diff
@@ -0,0 +1,44 @@
+run:
+  timeout: 10m
+  allow-parallel-runners: true
+
+linters-settings:
+  errcheck:
+    check-type-assertions: false
+    check-blank: true
+
+  maligned:
+    # print struct with more effective memory layout or not, false by default
+    suggest-new: true
+
+  unused:
+    go: "1.17"
+
+  lll:
+    # max line length, lines longer will be reported. Default is 120.
+    # '\t' is counted as 1 character by default, and can be changed with the tab-width option
+    line-length: 185
```

Comment:
Don't we want to actually set this to 120?

Reply:
We'd need to adjust a bunch of lines already and I'd just want to prevent adding crazy long lines here.

```
internal/stepfuncs/assess_funcs.go:33: line is 126 characters (lll)
func WaitUntilPodsAvailable(listOptions metav1.ListOptions, count int, wait time.Duration, tick time.Duration) features.Func {
internal/stepfuncs/assess_funcs.go:56: line is 125 characters (lll)
                terrak8s.WaitUntilServiceAvailable(t, &kubectlOpts, receiverMockServiceName, int(waitDuration), tickDuration)
internal/stepfuncs/assess_funcs.go:75: line is 136 characters (lll)
                                                return "", fmt.Errorf("couldn't find metric %q in received metrics", expectedMetricName)
internal/stepfuncs/assess_funcs.go:104: line is 125 characters (lll)
                terrak8s.WaitUntilServiceAvailable(t, &kubectlOpts, receiverMockServiceName, int(waitDuration), tickDuration)
helm_default_installation_test.go:41: line is 123 characters (lll)
                                k8s.WaitUntilSecretAvailable(t, ctxopts.KubectlOptions(ctx), "sumologic", 60, tickDuration)
helm_default_installation_test.go:74: line is 129 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-logs", releaseName),
helm_default_installation_test.go:112: line is 132 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-metrics", releaseName),
helm_default_installation_test.go:150: line is 131 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-events", releaseName),
helm_default_installation_test.go:235: line is 121 characters (lll)
                                        // For now let's take the newest metric sample only because it will have the most
helm_default_installation_test.go:244: line is 150 characters (lll)
                                                // https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/4508796836?check_suite_focus=true
helm_default_installation_test.go:261: line is 150 characters (lll)
                                                // https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/4508796836?check_suite_focus=true
helm_otc_metadata_installation_test.go:43: line is 128 characters (lll)
                                terrak8s.WaitUntilSecretAvailable(t, ctxopts.KubectlOptions(ctx), "sumologic", 60, tickDuration)
helm_otc_metadata_installation_test.go:79: line is 127 characters (lll)
                                                        fmt.Sprintf("app=%s-sumologic-otelcol-logs", ctxopts.HelmRelease(ctx)),
helm_otc_metadata_installation_test.go:121: line is 130 characters (lll)
                                                        fmt.Sprintf("app=%s-sumologic-otelcol-metrics", ctxopts.HelmRelease(ctx)),
helm_otc_metadata_installation_test.go:161: line is 131 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-events", releaseName),
helm_otc_metadata_installation_test.go:185: line is 124 characters (lll)
                                        daemonsets = terrak8s.ListDaemonSets(t, ctxopts.KubectlOptions(ctx), v1.ListOptions{
helm_otc_metadata_installation_test.go:246: line is 121 characters (lll)
                                        // For now let's take the newest metric sample only because it will have the most
helm_otc_metadata_installation_test.go:255: line is 150 characters (lll)
                                                // https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/4508796836?check_suite_focus=true
helm_otc_metadata_installation_test.go:259: line is 154 characters (lll)
                                                // For OTC metadata enrichment this is set to <RELEASE_NAME>-sumologic-otelcol-metrics-<POD_IN_STS_NUMBER>
helm_otc_metadata_installation_test.go:260: line is 121 characters (lll)
                                                // hence with longer time range the time series about a particular metric
helm_otc_metadata_installation_test.go:280: line is 150 characters (lll)
                                                // https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/4508796836?check_suite_focus=true
helm_traces_enabled_test.go:36: line is 128 characters (lll)
                                terrak8s.WaitUntilSecretAvailable(t, ctxopts.KubectlOptions(ctx), "sumologic", 60, tickDuration)
helm_traces_enabled_test.go:70: line is 129 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-logs", releaseName),
helm_traces_enabled_test.go:108: line is 132 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-metrics", releaseName),
helm_traces_enabled_test.go:146: line is 131 characters (lll)
                                                        LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-events", releaseName),
helm_traces_enabled_test.go:170: line is 128 characters (lll)
                                        daemonsets = terrak8s.ListDaemonSets(t, ctxopts.KubectlOptions(ctx), metav1.ListOptions{
helm_traces_enabled_test.go:184: line is 136 characters (lll)
                Assess("otelcol deployment is ready", func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
helm_traces_enabled_test.go:203: line is 137 characters (lll)
                                                DeploymentConditionMatch(&ds.Items[0], appsv1.DeploymentAvailable, corev1.ConditionTrue),
helm_traces_enabled_test.go:211: line is 137 characters (lll)
                Assess("otelagent daemonset is ready", func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
main_test.go:96: line is 136 characters (lll)
```